cron/data/publisher.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2021 Security Scorecard Authors
+//
+// Licensed under the Apache License, Vershandlern 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permisshandlerns and
+// limitathandlerns under the License.
+
+package data
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"sync"
+	"sync/atomic"
+
+	"gocloud.dev/pubsub"
+	_ "gocloud.dev/pubsub/gcppubsub" // Needed to link in GCP drivers.
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+var (
+	errorOpenTopic        = errors.New("pubsub.OpenTopic")
+	errorProtoJSONMarshal = errors.New("protojson.Marshal")
+	errorPublish          = errors.New("total errors when publishing")
+)
+
+type Publisher interface {
+	Publish(request *ScorecardBatchRequest) error
+	Close() error
+}
+
+func CreatePublisher(ctx context.Context, topicURL string) (Publisher, error) {
+	ret := publisherImpl{}
+	topic, err := pubsub.OpenTopic(ctx, topicURL)
+	if err != nil {
+		return &ret, fmt.Errorf("%w: %v", errorOpenTopic, err)
+	}
+	return &publisherImpl{
+		ctx:   ctx,
+		topic: topic,
+	}, nil
+}
+
+type topicInterface interface {
```

Comment:
Instead of `topicInterface` could be just `publisher`? And the method be `publish`.
 https://golang.org/doc/effective_go#interface-names
 
In golang By convention, one-method interface names are nouns denoting the doer of an action.